client/app/scripts/superdesk/upload/image-crop-directive.js

```diff
@@ -0,0 +1,132 @@
+
+                    cropSelect.push(cropImage.CropLeft);
+                    cropSelect.push(cropImage.CropTop);
+                    cropSelect.push(cropImage.CropRight);
+                    cropSelect.push(cropImage.CropBottom);
+
+                    return cropSelect;
+                }
+
+                // To adjust preview box as per aspect ratio
+                if (scope.aspectRatio) {
+                    if (scope.aspectRatio.toFixed(2) === '1.33') {
+                        rwidth = 300; rheight = 225;
+                    } else if (scope.aspectRatio.toFixed(2) === '1.78') {
+                        rwidth = 300; rheight = 169;
+                    } else {
+                        rwidth = 300; rheight = 300;
+                    }
+                } else {
+                    notify.error(gettext('sdImageCrop: attribute "aspect-ratio" is mandatory'));
+                    throw new Error('sdImageCrop: attribute "aspect-ratio" is mandatory');
+                }
+
+                var updateFunc = function(c) {
+                    cancelTimeout(c);
+                    updateTimeout = $timeout(updateScope(c), 300, false);
+                };
+
+                function cancelTimeout(event) {
+                    $timeout.cancel(updateTimeout);
+                }
+
+                function updateScope(c) {
+                    scope.$apply(function() {
+                        scope.cords = c;
+                        var rx = rwidth / scope.cords.w;
+                        var ry = rheight / scope.cords.h;
+                        showPreview('.preview-target-1', rx, ry, boundx, boundy, scope.cords.x, scope.cords.y);
+                    });
+                }
+
+                function showPreview(e, rx, ry, boundx, boundy, cordx, cordy) {
+                    $(e).css({
+                        width: Math.round(rx * boundx) + 'px',
+                        height: Math.round(ry * boundy) + 'px',
+                        marginLeft: '-' + Math.round(rx * cordx) + 'px',
+                        marginTop: '-' + Math.round(ry * cordy) + 'px'
+                    });
+                }
+
+                function validateConstraints(imgObj) {
+                    if (imgObj.width < minimumSize[0] || imgObj.height < minimumSize[1]) {
+                        scope.$apply(function() {
+                            notify.pop();
+                            notify.error(gettext('Sorry, but image must be at least ' + minimumSize[0] + 'x' + minimumSize[1]));
+                            scope.src = null;
+                            scope.$parent.preview.progress = null;
+                            throw new Error('sdImageCrop: Sorry, but image must be at least ' + minimumSize[0] + 'x' + minimumSize[1]);
+                        });
+                        return;
+                    }
+                }
+
+                scope.$watch('src', function(src) {
+                    elem.empty();
+                    if (src) {
```

Comment:
Maybe just reverse the condition (`if (!src) --> return`) to avoid indenting the whole function body.